src/UI/Components/Diagram/Context/EventWrapper.tsx

```diff
@@ -0,0 +1,189 @@
+    setCellToEdit,
+    setDictToDisplay,
+    looseEmbedded,
+    setLooseEmbedded,
+  } = useContext(CanvasContext);
+
+  /**
+   * Handles the event triggered when there are loose embedded entities on the canvas.
+   * The loose embedded entities are the entities that are not connected to the main entity. If there are any, they will be highlighted and deploy button will be disabled.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleLooseEmbeddedEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+    const eventData: { kind: EmbeddedEventEnum; id: string } = JSON.parse(
+      customEvent.detail,
+    );
+    const newSet = new Set(looseEmbedded);
+
+    if (eventData.kind === "remove") {
+      newSet.delete(eventData.id);
+    } else {
+      newSet.add(eventData.id);
+    }
+    setLooseEmbedded(newSet);
+  };
+
+  /**
+   * Handles the event triggered when the user wants to see the dictionary properties of an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleDictEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setDictToDisplay(JSON.parse(customEvent.detail));
+  };
+
+  /**
+   * Handles the event triggered when the user wants to edit an entity.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleEditEvent = (event): void => {
+    const customEvent = event as CustomEvent;
+
+    setCellToEdit(customEvent.detail);
+  };
+
+  /**
+   * Handles the event triggered when the user made update to the instance cell
+   * we need to assert first if the instance triggered the event is in the serviceOrderItems map,
+   * as the related instances aren't added to the serviceOrderItems map as we don't want to accidentally delete or edit them,
+   * yet they can exist on the canvas and be removed from it.
+   *
+   * @param {CustomEvent} event - The event object.
+   *
+   * @returns {void}
+   */
+  const handleUpdateServiceOrderItems = (event): void => {
+    const customEvent = event as CustomEvent;
+    const { cell, action } = customEvent.detail as {
+      cell: ServiceEntityBlock;
+      action: ActionEnum;
+    };
+
+    setServiceOrderItems((prev) => {
+      // related instances aren't added to the serviceOrderItems map, this condition is here to prevent situation where we try to remove the related instance from the canvas and it ends up here with status to delete it from the inventory
+      if (prev.has(String(cell.id)) || action === ActionEnum.CREATE) {
+        return updateServiceOrderItems(cell, action, prev);
+      }
+
+      return prev;
+    });
+  };
+
+  /**
+   * Handles updating the stencil state for the embedded entities.
+   * If the current count of the instances created from adequate stencil is more than or equal to the max count,, the adequate stencil element will become disabled.
```

Comment:
```suggestion
   * If the current count of elements created from a certain type of stencil is more than or equal to the max count, the corresponding stencil element will be disabled.
```
adequate means something a bit different than what you mean. Someone is adequate for a job, or a function. A tool is adequate for a task. You mean corresponding or matching. 
I'd also not use instances in the description considering that embedded entities aren't exactly instances.